system/security/plugins/jwtSecurity/jwtSecurity.cpp

```diff
@@ -0,0 +1,1121 @@
+            {
+                if (x->claim == "AllowWorkunitScopeView")
+                    userInfo.addWUScopePerm(x->scopePattern, SecAccess_Read, defaultWUAccess);
+                else if (x->claim == "AllowWorkunitScopeModify")
+                    userInfo.addWUScopePerm(x->scopePattern, SecAccess_Write, defaultWUAccess);
+                else if (x->claim == "AllowWorkunitScopeDelete")
+                    userInfo.addWUScopePerm(x->scopePattern, SecAccess_Full, defaultWUAccess);
+                else if (x->claim == "DenyWorkunitScopeView")
+                    userInfo.addWUScopePerm(x->scopePattern, SecAccess_Read, defaultWUAccess, true);
+                else if (x->claim == "DenyWorkunitScopeModify")
+                    userInfo.addWUScopePerm(x->scopePattern, SecAccess_Write, defaultWUAccess, true);
+                else if (x->claim == "DenyWorkunitScopeDelete")
+                    userInfo.addWUScopePerm(x->scopePattern, SecAccess_Full, defaultWUAccess, true);
+            }
+
+            // Add file scope permissions in alphabetical (and implicitly, length) order
+            std::sort(fileScopePerms.begin(), fileScopePerms.end());
+            for (CollectedScopePerms::const_iterator x = fileScopePerms.begin(); x != fileScopePerms.end(); x++)
+            {
+                if (x->claim == "AllowFileScopeAccess")
+                    userInfo.addFileScopePerm(x->scopePattern, SecAccess_Access, defaultFileAccess);
+                else if (x->claim == "AllowFileScopeRead")
+                    userInfo.addFileScopePerm(x->scopePattern, SecAccess_Read, defaultFileAccess);
+                else if (x->claim == "AllowFileScopeWrite")
+                    userInfo.addFileScopePerm(x->scopePattern, SecAccess_Write, defaultFileAccess);
+                else if (x->claim == "AllowFileScopeFull")
+                    userInfo.addFileScopePerm(x->scopePattern, SecAccess_Full, defaultFileAccess);
+                else if (x->claim == "DenyFileScopeAccess")
+                    userInfo.addFileScopePerm(x->scopePattern, SecAccess_Access, defaultFileAccess, true);
+                else if (x->claim == "DenyFileScopeRead")
+                    userInfo.addFileScopePerm(x->scopePattern, SecAccess_Read, defaultFileAccess, true);
+                else if (x->claim == "DenyFileScopeWrite")
+                    userInfo.addFileScopePerm(x->scopePattern, SecAccess_Write, defaultFileAccess, true);
+                else if (x->claim == "DenyFileScopeFull")
+                    userInfo.addFileScopePerm(x->scopePattern, SecAccess_Full, defaultFileAccess, true);
+            }
+
+            gUserCache.set(subject, userInfo);
+
+            verificationResult = true;
+        }
+        catch (const std::runtime_error& e)
+        {
+            // Most jwt-cpp exceptions are subclasses of std::runtime_error
+            PROGLOG("CJwtSecurityManager::verifyToken error: %s", e.what());
```

Comment:
I don't know if it would be the best, as I have not kept up on the logging changes.  Would DBGLOG be more appropriate for these?  Or a different function?